src/VisualStudio/Core/Def/Implementation/Options/RoamingVisualStudioProfileOptionPersister.cs

```diff
@@ -50,20 +58,33 @@ public RoamingVisualStudioProfileOptionPersister(IThreadingContext threadingCont
         {
             Contract.ThrowIfNull(globalOptionService);
 
-            _settingManager = (ISettingsManager)serviceProvider.GetService(typeof(SVsSettingsPersistenceManager));
             _globalOptionService = globalOptionService;
 
-            // While the settings persistence service should be available in all SKUs it is possible an ISO shell author has undefined the
-            // contributing package. In that case persistence of settings won't work (we don't bother with a backup solution for persistence
-            // as the scenario seems exceedingly unlikely), but we shouldn't crash the IDE.
-            if (_settingManager != null)
-            {
-                var settingsSubset = _settingManager.GetSubset("*");
-                settingsSubset.SettingChangedAsync += OnSettingChangedAsync;
-            }
+            var serviceContainer = serviceProvider.GetService<SVsBrokeredServiceContainer, IBrokeredServiceContainer>();
+            _serviceBroker = serviceContainer.GetFullAccessServiceBroker();
```

Comment:
I was under the impression that you should not store a service broker instance (as the underlying service could change b/w local and remote), but you could store a ServiceBrokerClient which will always return the correct service broker.

See https://github.com/dotnet/roslyn/pull/38653/files#diff-523a05cebc5ce23017413494ba44d97dR36 
https://microsoft.sharepoint.com/teams/VSSaaS/_layouts/15/Doc.aspx?sourcedoc={c98588b4-ecce-437a-8939-3130feae3489}&action=edit&wd=target%28VS%20Nexus%2FGeneral%20Info.one%7Ca3ea5b79-fc2e-433f-b15c-4059e72b1510%2F%F0%9F%94%8EConsuming%20brokered%20services%7C8a5fa795-046e-45fc-b62f-407d025969be%2F%29

Reply:
Ahh I see you re-query on availability changed.  that should be fine

Reply:
`IServiceBroker` is just fine to store as a field. It's also perfectly legit to use that interface on its own. `ServiceBrokerClient` is useful if you never need the service in between async method bodies (e.g. you don't subscribe to events nor rely on a stateful connection) because this wrapper class can cache the proxy to reduce latency from service acquisition each time you use it, while also managing lifetime if the service changes as connections are made or dropped. But `ServiceBrokerClient` must be disposed (`IServiceBroker` needn't be), and the "rentals" issued by the Client class should *not* be stored in a field. It's meant just for use as a local in an async method.